### PR TITLE
Some i2p changes

### PR DESCRIPTION
**Pull request by joelmo**

### CODE REVIEW

File: pkgs/tools/networking/i2p/default.nix [hunk 1]
```diff
@@ -36,6 +35,7 @@ stdenv.mkDerivation rec {
     description = "Applications and router for I2P, anonymity over the Internet";
     maintainers = [ stdenv.lib.maintainers.joelmo ];
     licenses = licenses.gpl2;
-    platforms = platforms.linux;
+    # TODO: support other systems, just copy appropriate lib/wrapper.. to $out
+    platforms = [ "x86_64-linux" ];
```
Comment (lethalman) [inline pkgs/tools/networking/i2p/default.nix]: But before this the check above was not to build for x86_64-linux, are you sure about this change?

File: pkgs/tools/networking/i2p/default.nix [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-
 stdenv.mkDerivation rec {
   name = "i2p-0.9.18";
   src = fetchurl {
     url = "https://github.com/i2p/i2p.i2p/archive/${name}.tar.gz";
     sha256 = "1hahdzvfh1zqb8qdc59xbjpqm8qq95k2xx22mpnhcdh90lb6xqnl";
   };
-  buildInputs = [ openjdk8 ant gcj gettext ];
+  buildInputs = [ ant gcj gettext ];
+  propagatedBuildInputs = [ openjdk8 ];
```
Comment (lethalman) [inline pkgs/tools/networking/i2p/default.nix]: Is this needed?

File: pkgs/tools/networking/i2p/default.nix (hunk 1 above)
Comment (joelmo) [inline pkgs/tools/networking/i2p/default.nix]: yes it should be only for 64-bit linux for now, I wrote the first assert wrong

File: pkgs/tools/networking/i2p/default.nix (hunk 2 above)
Comment (joelmo) [inline pkgs/tools/networking/i2p/default.nix]: I don't know, isn't there a possibility openjdk8 get garbadge collected otherwise?

Comment (lethalman) [inline pkgs/tools/networking/i2p/default.nix]: You've used it below with `${openjdk8}`, it can't be garbage collected.

**Comment by lethalman** [comment]
Looks fine, thanks.

**Pull request merged**